client/order/index.js

```diff
@@ -42,30 +44,57 @@ jQuery( function ( $ ) {
 			'wc-refunded' === this.value &&
 			'wc-refunded' !== originalStatus
 		) {
-			if ( ! canRefund ) {
-				alert(
-					__( 'Order cannot be refunded', 'woocommerce-payments' )
-				);
-				return;
-			}
-			if ( 0 >= refundAmount ) {
-				alert( __( 'Invalid Refund Amount', 'woocommerce-payments' ) );
+			renderRefundConfirmationModal(
+				originalStatus,
+				canRefund,
+				refundAmount
+			);
+		} else if (
+			'wc-cancelled' === this.value &&
+			'wc-cancelled' !== originalStatus
+		) {
+			if ( ! canRefund || 0 >= refundAmount ) {
```

Comment:
WDYT if we simplify the condition of showing the cancel modal a bit? Basically, move it into a separate `if` condition, and combine all conditions into that `if`. 

```suggestion
		} 
		
		if (
			'wc-cancelled' === this.value &&
			'wc-cancelled' !== originalStatus && 
			canRefund &&
			refundAmount > 0
```

Reply:
First set of condition = Did user click on cancelled
Second set of  condition = Is this order refundable.
Also, I feel like it reads easier when scanning the file due to similarity with the previous refund modal code, and I believe there are more modals to come.

Reply:
Thanks for providing more details. Not a blocker here for me to approve this PR.